### PR TITLE
Adds Elm front end example

### LINKED ISSUE #86: Elm UI with Tachyons
@finnhodgkin & @ZooeyMiller 

# TODO:

+ [x] Go through this tutorial and confirm you:
  + [x] Run the code on your `localhost`
  + [x] Test it on your mobile device connecting to your `localhost`
  + [x] Understand the code & tests (ask any questions if anything unclear)
+ [x] Create an endpoint for rendering the Elm UI
+ [x] Re-create the _current_ UI in Elm and Tachyons
+ [x] Hook up UI to Hapi Backend

### PR DESCRIPTION
**Pull request by ZooeyMiller**
Adds an Elm front end to the hapi/socket.io/redis back end, with tests and instructions of how to run it locally to the README.md 
fixes #86

### CODE REVIEW

File: elm/index.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <meta http-equiv="X-UA-Compatible" content="ie=edge">
+  <title>Chat</title>
```
Comment (nelsonic) [inline elm/index.html]: lack of favicon in `<head>` leads to `500` error:
image

File: elm/js/javascript.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+var node = document.getElementById('app');
+var storedName = localStorage.getItem('name');
+var socket = io();
```
Comment (nelsonic) [inline elm/js/javascript.js]: could _some_ of this have been included from the existing file (_i.e. avoid duplication_) ?

File: elm/src/Main.elm [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+
+type alias Message =
+    { n : String
+    , t : Time
+    , m : String
+    }
+
+
+
+-- Add a placeholder to messageInput so empty message submits prompt the user
```
Comment (nelsonic) [inline elm/src/Main.elm]: do you gents have https://github.com/avh4/elm-format installed?

File: server.js [hunk 4]
```diff
@@ -2,27 +2,60 @@ var Hapi = require('hapi');
...
+    {
+      method: 'GET',
+      path: '/js/javascript.js',
+      handler: {
+        file: './elm/js/javascript.js',
+      },
+    },
+    {
+      method: 'GET',
+      path: '/css/tachyons.min.css',
```
Comment (nelsonic) [inline server.js]: can we load `tachyons.min.css` from CDN rather than add another route for it?

**Comment by nelsonic** [comment]
Runs on `localhost`:
image

Tests pass (_as expected_):
image

_However_ seeing debug (_error_) in console:
image

Might be related to the fact that Redis is failing:
image

https://stackoverflow.com/questions/39809595/redis-server-failure-when-server-starts
image

File: elm/js/javascript.js (hunk 2 above)
Comment (ZooeyMiller) [inline elm/js/javascript.js]: I'm not really sure what you mean? As in copy/paste from the existing client.js or..? Would you like me to change any of it?

File: elm/src/Main.elm (hunk 3 above)
Comment (ZooeyMiller) [inline elm/src/Main.elm]: Yes we both do, are you referring to the line breaks with between the comments?
I just reduced all line breaks to 1, and on save elm format reformatted to the same as the quoted block here.

File: server.js (hunk 4 above)
Comment (ZooeyMiller) [inline server.js]: fixed by cbdba04

**Comment by ZooeyMiller** [comment]
@nelsonic I just started `redis-server`, then ran `npm start` and didn't get any of the errors you have gotten, now that you've upgraded redis are you getting them? 
I have more or less no experience with redis so if this needs more in depth work to fix then could we pair or something?

File: elm/index.html (hunk 1 above)
Comment (ZooeyMiller) [inline elm/index.html]: Thanks for sorting this for us, will be sure to add a favicon to any future projects.

**Comment by nelsonic** [comment]
@ZooeyMiller yeah, after upgrading Redis it works. 👍

**Comment by nelsonic** [comment]
What I did next:
```
rm -rf node_modules
npm install
npm run start-elm
```
Appears to work ...
image


_However_, still see the error in server console.
image

This issue has been raised in: https://github.com/dwyl/hapi-error/issues/49

**Comment by nelsonic** [comment]
`hapi-error@1.8.0` fixes this issue. https://github.com/dwyl/hapi-error/releases/tag/v1.8.0
trying it now on localhost.

**Comment by nelsonic** [comment]
No more errors on server. ✅  😥

**Pull request merged**